chat-client.py

```diff
@@ -0,0 +1,64 @@
+import socket
+import selectors
+import errno
+import sys
+import time
+
+HEADER_LENGTH = 10
+HOST = '127.0.0.1'
+PORT = 54321
+
+print('\nWelcome to the chat server!\n')
+time.sleep(1)
+print('Hit enter to send message or refresh message feed\n')
+time.sleep(1)
+print('Type "quit" to leave the chat\n')
+time.sleep(1)
+my_username = input('Enter username: ')
+
+# Set up client socket
+sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+sock.connect((HOST, PORT))
```

Comment:
Done. Should I be looking for specific exceptions though? I've read that bare `except:` statements are bad practise. Is my method of using sys.exit() ok for quitting program here?

Reply:
It seems to be a `socket.error` exception which is thrown in when an error with `connect()` occurs.

Yes, try to avoid "catch all" exceptions because it will hide exceptions that you might not have expected to happen. In those situations you would want it to crash the program so you are made aware of it and can then decide whether to fix the underlying issue or handle it appropriately.

Reply:
It seems `socket.error` has been superseded by `OSError` so have now handled for this